src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -118,9 +118,11 @@ public void OnStartActivity(Activity activity, object payload)
         if (textMapPropagator is not TraceContextPropagator)
         {
             var ctx = textMapPropagator.Extract(default, request, HttpRequestHeaderValuesGetter);
-
+            var activityContext = new ActivityContext(activity.TraceId, activity.ParentSpanId, activity.ActivityTraceFlags, activity.TraceStateString, true);
```

Comment:
Do we even need to create an `ActivityContext` struct? We should simply compare the extracted context's Trace, SpanId, and TraceState with `activity.TraceId`, `activity.ParentSpanId` and `activity.TraceStateString`.

Reply:
It seems that I made exactly that change earlier locally, but the commit was not pushed yet.
Shall I push that or do you want to keep the `ActivityContext` struct?